optimum/exporters/ipex/modeling_utils.py

```diff
@@ -296,18 +295,18 @@ def forward(
 
 # Adapted from https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L186
 class _IPEXLlamaMLP(nn.Module):
-    def __init__(self, module, config, distributed=False) -> None:
+    def __init__(self, module, config) -> None:
         if is_ipex_version("<", _IPEX_MINIMUM_VERSION_FOR_PATCHING):
             raise ImportError(
                 f"Only ipex version > {_IPEX_MINIMUM_VERSION_FOR_PATCHING} supports Linear2SiluMul, LinearAdd"
             )
         super().__init__()
         _setattr_from_module(self, module)
         self.config = config
-        self.distributed = distributed
         from intel_extension_for_pytorch.llm.modules import Linear2SiluMul, LinearAdd
 
-        if not self.distributed:
+        # LinearAllreduce and LinearLayer cannot use fused op LinearAdd
+        if module.down_proj.__class__.__name__ not in ["LinearAllreduce"]:
```

Comment:
when does this happen ? could you add a test for this ?

Reply:
Yes, but it requires importing deepspeed.

Reply:
> when does this happen ? could you add a test for this ?

In "DeepSpeed + transformers or optimum-intel" case, DeepSpeed will shard to model for tensor parallelism, and some of the sharded model's layer will be changed to "LinearAllReduce" to fuse linear and allreduce into one op.  So, in this case, layer name is not enough for us to judge whether we need do linear add fusion, that why we change the logic from op name only check to `op name + layer class` check.

W/ this DeepSpeed TP can enjoy optimum-intel's optimized model performance.

Reply:
thanks a lot for the explanation